lib/spack/spack/cmd/create.py

```diff
@@ -874,7 +874,7 @@ def get_name(name, url):
 
     result = simplify_name(result)
 
-    if not valid_fully_qualified_module_name(result):
+    if not re.match(r"^[a-z0-9-]+$", result):
```

Comment:
Should we extract this regex somewhere and give it a name?

Reply:
It's not used elsewhere, I don't really see the point of giving it a name since the error message is one line below. Note that this is the odd one out here, it's more strict than "package api v1" allowed in all other code paths.